### PR TITLE
Xdr enable disable incident closure mirroring

### PR DESCRIPTION
**Pull request by noydavidi**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-10563)

## Description
Added the ability to independently enable/disable automatic incident closure mirroring between Cortex XDR and XSOAR.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 1]
```diff
@@ -55,6 +55,26 @@ configuration:
...
+  defaultvalue: 'true'
+  display: Close Mirrored XSOAR Incident
+  name: close_cortex_incident
+  type: 8
+  hidden:
+  - marketplacev2
+  section: Collect
+  advanced: true
+  required: false  
+- additionalinfo: When selected, closing the Cortex XSOAR incident is mirrored in Cortex XDR
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:68]:
```suggestion
- additionalinfo: When selected, closing the Cortex XSOAR incident is mirrored in Cortex XDR.
```

**Comment by ShirleyDenkberg** [comment]
@maimorag @Shellyber Doc review completed.

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 2]
```diff
@@ -7,33 +7,38 @@ This integration was integrated and tested with version 2.6.5 of Cortex XDR - IR
...
+    | Custom close-reason mapping (XDR -&gt; XSOAR mirrored incident. Overwrites default close-reason mapping defined by Cortex XSOAR) | Define how to close the mirrored incidents from Cortex XDR into Cortex XSOAR with a custom close reason mapping. Enter a comma-separated list of close reasons \(acceptable format \{Cortex XDR close reason\}=\{Cortex XSOAR close reason\}\) to override the default close reason mapping defined by Cortex XSOAR. Note that the mapping must be configured accordingly with the existing close reasons in Cortex XSOAR and Cortex XDR. Not following this format will result in closing the incident with a default close reason. Example: “Known Issue=Resolved, Duplicate Incident=Other". Refer to ../README.md for possible close-reasons - \`XDR Incident Mirroring\`. | False |
+    | API Key ID |  | False |
+    | API Key |  | False |
```
Comment (Shellyber) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:25]: It seems like it exists twice. 
Please check that it looks ok - https://github.com/demisto/content/blob/3446769845fa66ae5c7964674901a0293acf179c/Packs/CortexXDR/Integrations/CortexXDRIR/README.md
Verify the only change is the added params

**Review by Shellyber — CHANGES_REQUESTED** [review]
Great work.
See my comments.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 3]
```diff
@@ -55,6 +55,26 @@ configuration:
   - Incoming
   - Outgoing
   - Both
+- additionalinfo: When selected, closing the Cortex XDR incident is mirrored in Cortex XSOAR.
+  defaultvalue: 'true'
```
[the remark this answers is not in the document]
Comment (noydavidi) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:59]: Yes :)

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 4]
```diff
@@ -55,6 +55,26 @@ configuration:
...
+  display: Close Mirrored XSOAR Incident
+  name: close_cortex_incident
```
[the remark this answers is not in the document]
Comment (noydavidi) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml]: You are right. I changed it now

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md (hunk 2 above)
Comment (noydavidi) [reply Packs/CortexXDR/Integrations/CortexXDRIR/README.md]: Deleted the duplicated fields and compared the old README with the new one.

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 5]
```diff
@@ -7,33 +7,35 @@ This integration was integrated and tested with version 2.6.5 of Cortex XDR - IR
...
+    | Custom lose-reason mapping for mirrored **XDR -> XSOAR** incidents. | Define how to close the mirrored incidents from Cortex XDR into Cortex XSOAR with a custom close reason mapping. Enter a comma-separated list of close reasons (acceptable format {XDR close reason}={XSOAR close reason}) to override the default close reason mapping defined by Cortex XSOAR. Note that the mapping must be configured accordingly with the existing close reasons in Cortex XSOAR and Cortex XDR. Not following this format will result in closing the incident with a default close reason. Example: “Known Issue=Resolved". Default: “Known Issue=Other,Duplicate Incident=Duplicate,False Positive=False Positive,True Positive=Resolved,Security Testing=Other,Other=Other,Auto=Resolved". Refer to the integration documentation for possible close-reasons (`XDR Incident Mirroring, sec. 7`). | False |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:23]:
```suggestion
    | Custom lose-reason mapping for mirrored **XDR -> XSOAR** incidents. | Define how to close the mirrored incidents from Cortex XDR into Cortex XSOAR with a custom close reason mapping. Enter a comma-separated list of close reasons (acceptable format {Cortex XDR close reason}={Cortex XSOAR close reason}) to override the default close reason mapping defined by Cortex XSOAR. Note that the mapping must be configured accordingly with the existing close reasons in Cortex XSOAR and Cortex XDR. Not following this format will result in closing the incident with a default close reason. Example: “Known Issue=Resolved". Default: “Known Issue=Other,Duplicate Incident=Duplicate,False Positive=False Positive,True Positive=Resolved,Security Testing=Other,Other=Other,Auto=Resolved". Refer to the integration documentation for possible close-reasons (`XDR Incident Mirroring, sec. 7`). | False |
```
It says to refer to the integration documentation....
What integration documentation?

File: Packs/CortexXDR/Integrations/CortexXDRIR/README.md [hunk 6]
```diff
@@ -7,33 +7,35 @@ This integration was integrated and tested with version 2.6.5 of Cortex XDR - IR
...
+    | Only fetch starred incidents |  | False |
+    | Starred incidents fetch window | Starred fetch window timestamp \(&lt;number&gt; &lt;time unit&gt;, e.g., 12 hours, 7 days\). Fetches only starred incidents within the specified time range. | False |
+    | First fetch timestamp (&lt;number&gt; &lt;time unit&gt;, e.g., 12 hours, 7 days) |  | False |
+    | Sync Incident Owners | For Cortex XSOAR version 6.0.0 and above. If selected, for every incident fetched from Cortex XDR to Cortex XSOAR, the incident owners will be synced. Note that once this value is changed and synchronized between the systems, additional changes will not be reflected. For example, if you change the owner in Cortex XSOAR, the new owner will also be changed in Cortex XDR. However, if you now change the owner back in Cortex XDR, this additional change will not be reflected in Cortex XSOAR. In addition, for this change to be reflected, the owners must exist in both Cortex XSOAR and Cortex XDR. | False |
+    | Trust any certificate (not secure) |  | False |
+    | Use system proxy settings |  | False |
+    | Prevent Only Mode | Whether the XDR tenant mode is prevent only | False |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/Integrations/CortexXDRIR/README.md:31]:
```suggestion
    | Prevent Only Mode | Whether the Cortex XDR tenant mode is prevent only .| False |
```
What is prevent only mode?

File: Packs/CortexXDR/ReleaseNotes/6_1_83.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Palo Alto Networks Cortex XDR - Investigation and Response
+
+- Added the *Close Mirrored Cortex XDR Incident* and *Close Mirrored XSOAR Incident* parameters to enable automatic closure of XDR and XSOAR incidents during mirroring.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXDR/ReleaseNotes/6_1_83.md:6]:
```suggestion
Added the *Close Mirrored Cortex XDR Incident* and *Close Mirrored XSOAR Incident* parameters to enable automatic closure of Cortex XDR and Cortex XSOAR incidents during mirroring.
```

**Comment by ShirleyDenkberg** [comment]
@noydavidi As requested, I reviewed the doc again.

**Review by Shellyber — APPROVED** [review]
Nice!

**Comment by noydavidi** [comment]
Currently, "Test Playbook - Cortex XDR Malware - Incident Enrichment" for XDR is failing.
After an investigation the TPB is not failing because of the changes in the code. **The TPB is also failing in Nightly.**

**Pull request merged**